tests/test_configurable_lookback_window.py

```diff
@@ -0,0 +1,85 @@
+"""Test tap configurable properties. Specifically the lookback_window"""
+import re
+import os
+from datetime import datetime as dt
+from datetime import timedelta
+
+from tap_tester import menagerie, connections, runner
+
+from base import BaseTapTest
+
+
+class ConversionWindowBaseTest(BaseTapTest):
+    """
+    Test tap's sync mode can execute with valid lookback_window values set.
+    Validate setting the lookback_window configurable property.
+    Test Cases:
+    Verify connection can be created, and tap can discover and sync with a lookback window
+    when passed in config else takes default value.
+    """
+    lookback_window = ''
+
+    def name(self):
+        return f"tt_stripe_lookback_window_{self.lookback_window}"
+
+    def get_properties(self):
+        """Configurable properties, with a switch to override the 'start_date' property"""
+        return_value = {
+            'start_date':dt.strftime(dt.utcnow() - timedelta(seconds=600), self.START_DATE_FORMAT),
+            'lookback_window': self.lookback_window,
+            'account_id': os.getenv('TAP_STRIPE_ACCOUNT_ID'),
+            'client_secret': os.getenv('TAP_STRIPE_CLIENT_SECRET')
+        }
+        return return_value
+
+    def run_test(self):
+        """
+        Testing that basic sync functions without Critical Errors when
+        a valid lookback_window is set.
+        """
+        print("Configurable Properties Test (lookback_window)")
+
+        conn_id = connections.ensure_connection(self)
+
+        streams_to_test = {'balance_transactions'}
+
+        # Run a discovery job
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Perform table and field selection...
+        core_catalogs = [catalog for catalog in found_catalogs
+                         if catalog['stream_name'] in streams_to_test]
+
+        # select all fields for core streams and...
+        self.select_all_streams_and_fields(conn_id, core_catalogs, select_all_fields=True)
+
+        # set state to ensure conversion window is used
+        today_datetime = int(dt.utcnow().timestamp())
+        
+        initial_state = {
+            'currently_syncing': None,
+            'bookmarks': {stream: {'created': today_datetime}
+                          for stream in streams_to_test
+                          if stream == 'balance_transactions'}
+        }
+        initial_state = {'currently_syncing': None, 'bookmarks': {'balance_transactions': {"created": today_datetime}}}
```

Comment:
There are duplicate declarations for  initial_state. 
Also I am not sure that a timestamp should be a valid bookmark format for a datetime value. Not really a functional issue, and if that is the existing behavior that I don't think it's worth altering the tap, but for consistency across taps in the future I think we prefer bookmarked datetimes to be formatted as ISO 8601/RFC 3339 utc.

Reply:
- Removed the duplicate declaration. 
- Stripe uses timestamp as a format for the bookmark value.
- Sure, we'll take care of this in the future, if a tap has bookmark as timestamp we'll convert it to ISO 8601/RFC 3339 utc.